### ISSUE TITLE
Backend

### ISSUE THREAD

**Issue by dashed** [5a1t/Mergr#1]
# Tech stack

- rust
- go-lang (maybe?)

**Comment by 5a1t** [comment]
Have you used rust? No experience with it but I have wanted to try it.

**Comment by dashed** [comment]
Yep. I'm using rust, and currently watching closely to performance benefits: https://github.com/tailhook/rotor

**Comment by synth3tk** [comment]
I've also wanted to try Rust, but never got around to it. Any good examples of this in our use case out in the wild?

As much as I think Node would actually work really well here, I feel we should get outside of comfort zones, as a lot of people did when Parrot was written.

**Comment by synth3tk** [comment]
If chat is implemented, are we pretty much unanimous on the IRC idea?

**Comment by 5a1t** [comment]
I wouldn't say we are unanimous yet, we should experiment if it's a promising idea.  It may very well be that it's way more work to interface with IRC than it is to just store channels and messages ourselves.

**Comment by dashed** [comment]
rust + websockets might be the way to go. I suggested rust because, as evident from Reddit having to shutdown Robin prematurely, we can squeeze out a lot of performance (c-like performance) and easily solve the C10k problem. Also, rust would minimize debugging (especially runtime bugs) in the long run.

**Comment by 5a1t** [comment]
Is there a Rust web package that you prefer?

**Comment by dashed** [comment]
I created some RESTful flashcard app using rust (as part of learning rust) using this https://github.com/iron/iron

Maybe something like this might be good for us: https://github.com/cyderize/rust-websocket

---

By the way, this tutorial series is incredibly relevant: https://nbaksalyar.github.io/2015/11/09/rust-in-detail-2.html

**Comment by Daegalus** [comment]
I am down for Rust, or Scala. So this all works for me. Ive been wanting a good reason to mess with Rust
